inst/stan/functions/zero_truncated_normal.stan

```diff
@@ -0,0 +1,13 @@
+real zero_truncated_normal_lpdf(vector y, real mu, real sigma) {
```

Comment:
I've been reading more about this and I think the lower bound only version might actually be functionality equivalent. Still not 💯  but pretty sure.

Reply:
I don't think it hurts to have it explicitly implemented but could potentially be fractionally more performant to just use a lower bound.

Reply:
Let's stay with the explicit truncation for now, as discussed :)